src/tutorials/data/queries.md

```diff
@@ -0,0 +1,365 @@
+
+Here, we explain how you can index fields to efficiently query documents. If you are interested on why indexing is important and how your performances can dramatically vary, you can go to the advanced explanations [here](./advanced.md#indexes-performances-and-design).
+
+With mango queries, it is required to index documents fields that are necessary to run the queries, either because they are involved in the selector or the sort.
+  
+### Index with cozy-client
+
+With cozy-client, the developer does not have to deal with the index creation: it is expressed through `indexFields`:
+
+```javascript
+const queryDef = client
+  .find("io.cozy.todos")
+  .where({
+    "category": "sport"
+  })
+.indexFields(["category"])
+```
+
+Thanks to this, cozy-client will automatically create an index on `category` before running the query for the first time and explicity tell CouchDB to use it.
+
+💡 In this example, the `indexField` method is not mandatory: cozy-client is smart enough to automatically index `category`  as it is involved in the selector. However, this automatic indexing only works for simple queries, thus, it is encouraged to explicity declare which fields to index.
+
+⚠️ CouchDB evaluates if a new document should be indexed by checking if its fields match an existing index. It means that **all the indexed fields must exist in the document** to index it. It can be problematic for queries that needs to check if a particular field exists or not. We detail this behaviour and workarounds in [this section](./advanced.md#indexes-why-is-my-document-not-being-retrieved).
+
+ℹ️ If no index is used to run the query, the query response will include a `warning: no matching index found, create an index to optimize query time`. It means that your index is not properly defined, probably because some fields involved in the `where`  selector are not indexed. Note that if you use a `sortBy`, you also need to index the fields involved, as explained in [this section](#sort-data).
+
+### Index update
+
+CouchDB updates the Mango indexes when the data is read, but not on writes. This implies that if one creates an index and inserts documents, the index will never be updated until a `find` query on these documents is performed. 
+
+⚠️ This implies that a query performed after many writes can take a certain time to complete, as it will update the index before returning any result. Likewise, the first query after an index creation will include the index build. Hence, developers should keep this behaviour in mind when designing applications and typically might need to handle query latency. 
+
+ℹ️ Starting from CouchDB 3.0.0, a [background indexing](https://docs.couchdb.org/en/3.0.0/config/indexbuilds.html) is implemented that aims to avoid this latency by forcing the index to update in background after writes operations.
```

Comment:
```suggestion
ℹ️ Starting from CouchDB 3.0.0, [background indexing](https://docs.couchdb.org/en/3.0.0/config/indexbuilds.html) is implemented. It aims to avoid this latency by forcing the index to update in background after writes.
```